File: proxy/tests/script_engine/trailer_has.lua
Extensions = {
	{
		request = function(flow)
			if flow.request.trailers:has("X-trailer1") then
				flow.request.body.text = "has"
			end
		end,
		response = function(flow)
			if flow.response.trailers:has("X-trailer1") then
				flow.response.body.text = "has"
			end
		end,
	},
}

File: proxy/tests/script_engine/trailer_has.py
from roxy import Extension


class HasTrailer(Extension):
    def request(self, flow):
        if flow.request.trailers.has("X-trailer1"):
            flow.request.body.text = "has"

    def response(self, flow):
        if flow.response.trailers.has("X-trailer1"):
            flow.response.body.text = "has"


Extensions = [HasTrailer()]
